### PR TITLE
v1 API documentation page template

### PR DESCRIPTION
**Pull request by juliawu**
Added a template API page file, along with some javascript (custom syntax highlighting) and CSS to make things look nice. 

Screenshot 2022-04-19 3 45 28 PM
Screenshot 2022-04-19 2 08 25 PM
Screenshot 2022-04-19 2 08 37 PM
Screenshot 2022-04-19 2 08 55 PM

### CODE REVIEW

File: _sass/api_documentation.scss [hunk 1]
```diff
@@ -0,0 +1,100 @@
+/**
+Custom CSS for API Method Documentation pages
+
+See api/rest/rest_api_template.md for an example of how these are used.
+**/
+
+//custom REQUIRED label for parameters
+required-tag {
```
Comment (shifucun) [inline _sass/api_documentation.scss:8]: Nit: can try to use some formatter to format this file. For example, can make the indent 2 spaces to match other files and align curly bracket.

File: assets/js/syntax_highlighting.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/*
+Custom syntax highlighting for REST api request signatures, for use on documentation pages.
```
Comment (shifucun) [inline assets/js/syntax_highlighting.js:2]: api -> API

**Review by shifucun — APPROVED** [review]
Thanks for sending this, nice and clean! Please wait for others' comments before check this in.

**Review by beets — APPROVED** [review]
this is nice! same comments regarding style as bo. a tool like prettier should help with this.

**Pull request merged**